AVFoundation-Combine/Video Player/VideoPlayerViewController.swift

```diff
@@ -0,0 +1,163 @@
+        avPlayerViewController.showsPlaybackControls = false
+        guard let contentOverlayView = avPlayerViewController.contentOverlayView else {
+            fatalError("`contentOverlayView` is required.")
+        }
+        contentOverlayView.addSubviewAndFillBounds(customUI)
+        customUI.translatesAutoresizingMaskIntoConstraints = false
+        customUI.playbackButton.addTarget(self, action: #selector(togglePlayback), for: .touchUpInside)
+        customUI.progressSlider.addTarget(self, action: #selector(onSliderThumbTouchedDown), for: .touchDown)
+        customUI.progressSlider.addTarget(self, action: #selector(onSliderThumbTouchedUp), for: .touchUpOutside)
+        customUI.progressSlider.addTarget(self, action: #selector(onSliderThumbTouchedUp), for: .touchUpInside)
+    }
+    
+    // MARK: UI Actions
+    
+    // MARK: Scubber
+    
+    @objc private func onSliderThumbTouchedDown() {
+        isProgressSliderScrubbing = true
+    }
+    
+    @objc private func onSliderThumbTouchedUp() {
+        avPlayerViewController.player?.seek(to: CMTime(seconds: Double(customUI.progressSlider.value), preferredTimescale: 1)) {[weak self] _ in
+            self?.isProgressSliderScrubbing = false
+        }
+    }
+    
+    // MARK: Play / Pause button
+    
+    @objc private func togglePlayback() {
+        avPlayerViewController.player?.rate == 0.0 ? avPlayerViewController.player?.play() : avPlayerViewController.player?.pause()
+    }
+    
+    // MARK: Video Player setup
+    
+    private func setup(_ player: AVPlayer) {
+        player.currentItemPublisher()
+            .compactMap { $0 }
+            .sink {[weak self] item in
+                self?.setup(item)
+                self?.avPlayerViewController.player?.play()
+            }
+            .store(in: &subscriptions)
+        
+        player.playheadProgressPublisher()
+            .filter {progress in
+                !self.isProgressSliderScrubbing
+            }
+            .map { Float($0) }
+            .assign(to: \.value, on: customUI.progressSlider)
+            .store(in: &subscriptions)
+        
+        let rateStream = player.ratePublisher().share()
+        
+        rateStream.receive(on: DispatchQueue.main)
+            .map { $0 == 0.0 ? "Play" : "Pause" }
+            .assign(to: \.accessibilityLabel, on: customUI.playbackButton)
+            .store(in: &subscriptions)
+        
+        rateStream.receive(on: DispatchQueue.main)
+            .map { $0 == 0.0 ? UIImage(named: "Play") : UIImage(named: "Pause") }
+            .sink {[weak self] image in
+                self?.customUI.playbackButton.setImage(image, for: .normal)
+            }
+            .store(in: &subscriptions)
+        
+        avPlayerViewController.player = player
+        player.replaceCurrentItem(with: AVPlayerItem(url: videoURL))
+    }
+    
+    private func setup(_ item: AVPlayerItem) {
+        
+        item.isPlaybackLikelyToKeepUpPublisher()
```

Comment:
I feel like this could use a `removeDuplicates` operator to only communicate distinct changes.